react/components/CreateOrganizationModal/CreateOrganizationModal.tsx

```diff
@@ -0,0 +1,378 @@
+import React, { useEffect, useState } from 'react'
+import { useMutation, useQuery } from 'react-apollo'
+import { FormattedMessage, useIntl } from 'react-intl'
+import { useToast } from '@vtex/admin-ui'
+import { Button, Input, Modal, Spinner } from 'vtex.styleguide'
+import {
+  AddressContainer,
+  AddressForm,
+  AddressRules,
+  CountrySelector,
+  PostalCodeGetter,
+} from 'vtex.address-form'
+import { StyleguideInput } from 'vtex.address-form/inputs'
+import { addValidation } from 'vtex.address-form/helpers'
+import { useRuntime } from 'vtex.render-runtime'
+
+import { getEmptyAddress, isValidAddress } from '../../utils/addresses'
+import { validatePhoneNumber } from '../../modules/formValidators'
+import CustomFieldInput from '../../admin/OrganizationDetailsCustomField'
+import { organizationMessages as messages } from '../../admin/utils/messages'
+import GET_LOGISTICS from '../../graphql/getLogistics.graphql'
+import CREATE_ORGANIZATION from '../../graphql/createOrganization.graphql'
+import GET_B2B_CUSTOM_FIELDS from '../../graphql/getB2BCustomFields.graphql'
+import {
+  INITIAL_FETCH_LIST_OPTIONS,
+  useOrganizationsList,
+} from '../../organizations/hooks'
+
+interface Props {
+  open: boolean
+  onOpenChange: (open: boolean) => void
+}
+
+const CreateOrganizationModal: React.FC<Props> = ({ open, onOpenChange }) => {
+  const { formatMessage } = useIntl()
+  const {
+    culture: { country },
+  } = useRuntime()
+
+  const showToast = useToast()
+
+  const [loadingState, setLoadingState] = useState(false)
+
+  const [newOrganizationName, setNewOrganizationName] = useState('')
+  const [newCostCenterName, setNewCostCenterName] = useState('')
+  const [newCostCenterPhoneNumber, setNewCostCenterPhoneNumber] = useState('')
+  const [
+    newCostCenterBusinessDocument,
+    setNewCostCenterBusinessDocument,
+  ] = useState('')
+
+  const [
+    newCostCenterStateRegistration,
+    setNewCostCenterStateRegistration,
+  ] = useState('')
+
+  const [newCostCenterAddressState, setNewCostCenterAddressState] = useState(
+    addValidation(getEmptyAddress(country))
+  )
+
+  const { data: logisticsData } = useQuery(GET_LOGISTICS, { ssr: false })
+  const [createOrganization] = useMutation(CREATE_ORGANIZATION)
+  const { refetch } = useOrganizationsList()
```

Comment:
This is the only part not directly copied from OrganizationsList. Since this logic is not in the same file that fetches/refetches the organizations data anymore, I had to create this new `useOrganizationsList` hook to achieve this. When `refetch` is called here, `OrganizationsList` is automatically updated with the newly fetched data (that's React's automatic rerendering magic on state updates).